core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -875,6 +878,82 @@ public static String translateWithCharset(String s, String transcodingName) {
     }
   }
 
+  /** SQL {@code PARSE_URL(urlStr, partToExtract [, keyToExtract] )} function. */
+  public static @Nullable String parseUrl(@Nullable String urlStr,
+      @Nullable String partToExtract) {
+    if (urlStr == null || partToExtract == null) {
+      return null;
+    }
+
+    URI uri;
+    try {
+      uri = new URI(urlStr);
+    } catch (URISyntaxException e) {
+      return null;
+    }
+
+    String extractValue;
+    PartToExtract part;
+    try {
+      part = PartToExtract.valueOf(partToExtract);
+    } catch (IllegalArgumentException e) {
+      return null;
+    }
+
+    switch (part) {
+    case HOST:
+      extractValue = uri.getHost();
+      break;
+    case PATH:
+      extractValue = uri.getRawPath();
+      break;
+    case QUERY:
+      extractValue = uri.getRawQuery();
+      break;
+    case REF:
+      extractValue = uri.getRawFragment();
+      break;
+    case PROTOCOL:
+      extractValue = uri.getScheme();
+      break;
+    case FILE:
+      if (uri.getRawQuery() != null) {
+        extractValue = uri.getRawPath() + "?" + uri.getRawQuery();
+      } else {
+        extractValue = uri.getRawPath();
+      }
+      break;
+    case AUTHORITY:
+      extractValue = uri.getRawAuthority();
+      break;
+    case USERINFO:
+      extractValue = uri.getRawUserInfo();
+      break;
+    default:
+      return null;
```

Comment:
Is this the right behavior? Or should it throw an error?

Reply:
Oh I see you have a test for so I guess it is right

Reply:
Thanks for remind, I think this can be improved to `break;`, I will update it soon.
Note that since we have handled `IllegalArgumentException` before, here we don't need to handle mismatched enum value again, and can safely use `break;`.